### DISCUSSION TITLE
Override dependency version for packages in workspaces

### DISCUSSION THREAD

**Discussion by pete-willard** [pnpm/pnpm#8388] [Q&A]
My setup is as below:

```
apps:
|-frontend (devDependencies: { 'eslint-fe': 'workspace:*' })
|-backend (devDependencies: { 'eslint-be': 'workspace:*' })

packages:
|-eslint-fe (devDependencies: { 'eslint-plugin': '3.x' })
|-eslint-be (devDependencies: { 'eslint-plugin': '2.x' })
|-...
```

- monorepo with FE/BE apps and several shared packages, single lock-file, turbo,  workspaces, shamefully-hoisted
- each app has its eslint config separated to a dedicated package (both use different plugins and config formats)
- one eslint plugin is used in both app eslint configs but in two different versions, 2.x and 3.x
- when deps are installed the root node_modules contains the 2.x version of the plugin and this makes linting the frontend app fail because it uses the 2.x binary instead of the 3.x

Is there a way to solve this? How do I force the frontend app to use 'eslint-plugin': '3.x'? From what I've read overrides only work in the root package.json and from what I can gather I need to override a dependency version for a specific workspace.

**Comment by zkochan** [comment]
This looks like an ESLint issue and according to [this](https://github.com/pnpm/pnpm/issues/8378) it should be fixed in ESLint v9.

**Reply by pete-willard** [reply]
Thanks for finding the time to check up and answer @zkochan! In the meantime I managed to work around this by using [aliases](https://pnpm.io/aliases).
